### ISSUE TITLE
TLS private key is world-readable by default

### ISSUE THREAD

**Issue by alvinhochun** [mailcow/mailcow-dockerized#2428]
**Describe the bug, try to make it reproducible**

With the installation method outlined in the doc, the cert and key files have mode 644 and is readable by other users.

```
[foo@bar ~]$ cat /opt/mailcow-dockerized/data/assets/ssl/key.pem
-----BEGIN RSA PRIVATE KEY-----
*** snip ***
-----END RSA PRIVATE KEY-----
[foo@bar ~]$ namei -nom /opt/mailcow-dockerized/data/assets/ssl/key.pem
f: /opt/mailcow-dockerized/data/assets/ssl/key.pem
 dr-xr-xr-x root root /
 drwxr-xr-x root root opt
 drwxr-xr-x root root mailcow-dockerized
 drwxr-xr-x root root data
 drwxr-xr-x root root assets
 drwxr-xr-x root root ssl
 -rw-r--r-- root root key.pem
```

**System information and quick debugging**

n/a

Further information (where applicable):

I don't know if the included ACME client sets the permission properly since I don't use it. However, if a user follows the instructions in https://mailcow.github.io/mailcow-dockerized-docs/firststeps-rp/#optional-post-hook-script-for-non-mailcow-acme-clients to deploy their own certs, the private key can end up being exposed to all users on the server.

At the very least the dir `data/assets/ssl` should have mode 700 by default.

**Comment by andryyy** [comment]
No, you don't set 700 on that whole directory. You need to set 600 on the keys, yes

**Comment by andryyy** [comment]
Fixed in c77368ee705db354bfd1b291c3b0f12e88d8033f
